gobblin-core/src/main/java/gobblin/source/extractor/extract/sftp/SftpFsHelper.java

```diff
@@ -162,6 +175,21 @@ public void connect()
   }
 
   /**
+   * In a map reduce environment the method sets the private key from a file on HDFS
+   */
+  private void setIdentityFromDefaultFileSystem(String privateKey, JSch jsch) throws IOException, JSchException {
+    FSDataInputStream privateKeyStream = null;
+    try {
+      FileSystem fs = FileSystem.get(new Configuration());
+      privateKeyStream = fs.open(new Path(privateKey));
+      byte[] bytes = new byte[1024];
```

Comment:
Is the length of the private key guaranteed to be 1024 bytes? It may be easier to use one of these approaches http://stackoverflow.com/questions/6058003/elegant-way-to-read-file-into-byte-array-in-java to convert a file to a byte array.